runtime/compiler/runtime/RelocationRecord.cpp

```diff
@@ -2157,31 +2157,31 @@ TR_RelocationRecordInlinedMethod::fixInlinedSiteInfo(TR_RelocationRuntime *reloR
 void
 TR_RelocationRecordInlinedMethod::preparePrivateData(TR_RelocationRuntime *reloRuntime, TR_RelocationTarget *reloTarget)
    {
-   J9Method *ramMethod = NULL;
-   bool failValidation = true;
+   TR_OpaqueMethodBlock *ramMethod = NULL;
+   bool inlinedSiteIsValid = inlinedSiteValid(reloRuntime, reloTarget, &ramMethod);
 
-   if (validateClassesSame(reloRuntime, reloTarget, ((TR_OpaqueMethodBlock **)&ramMethod)) &&
-       !reloRuntime->options()->getOption(TR_DisableCHOpts))
+   if (ramMethod)
       {
       // If validate passes, no patching needed since the fall-through path is the inlined code
-      fixInlinedSiteInfo(reloRuntime, reloTarget, (TR_OpaqueMethodBlock *) ramMethod);
-      failValidation = false;
+      fixInlinedSiteInfo(reloRuntime, reloTarget,ramMethod);
```

Comment:
I had a concern with this approach where the bug exposing this problem was due to sharing a symbol reference between two different inlined sites. So it seemed like a safer fix would be to either 1) not share the symbol reference between different inlined sites, or 2) only share "up": i.e. never use the symbol reference for an "inner" method in a reference in an "outer" method.

However, while chatting with Irwin, I convinced myself that 1 isn't desirable since it introduces a difference between AOT and JIT that would pre-req alias analysis to overcome, and 2 isn't always possible because you could have two "sibling" inlines that end up sharing a symbol reference between them. In this case, there's no "right" symbol reference to necessarily choose.

So, I like this solution best of the alternatives, I think. Thanks for putting it together!